src/builder/pdb.rs

```diff
@@ -0,0 +1,311 @@
+/// before being able to construct the [`PodDisruptionBudget`]:
+///
+/// 1. `metadata`
+/// 2. `selector`
+/// 3. Either `minAvailable` or `maxUnavailable`
+///
+/// Both `metadata` and `selector` will be set by [`PodDisruptionBudgetBuilder::new_with_role`].
+pub struct PodDisruptionBudgetBuilder<ObjectMeta, LabelSelector, PodDisruptionBudgetConstraint> {
+    metadata: ObjectMeta,
+    selector: LabelSelector,
+    /// Tracks wether either `maxUnavailable` or `minAvailable` is set.
+    constraint: Option<PodDisruptionBudgetConstraint>,
+}
+
+/// We intentionally only support fixed numbers, no percentage, see ADR 30 on Pod disruptions for details.
+/// We use u16, as [`IntOrString`] takes an i32 and we don't want to allow negative numbers. u16 will always fit in i32.
+#[derive(Debug)]
+pub enum PodDisruptionBudgetConstraint {
+    MaxUnavailable(u16),
+    MinAvailable(u16),
+}
+
+impl PodDisruptionBudgetBuilder<(), (), ()> {
+    pub fn new() -> Self {
+        PodDisruptionBudgetBuilder::default()
+    }
+
+    /// This method populates `metadata` and `selector` from the give role (not roleGroup!).
+    ///
+    /// The parameters are the same as the fields from [`crate::labels::ObjectLabels`]:
+    /// * `owner` - Reference to the k8s object owning the PDB, this should be e.g. `HdfsCluster` or `TrinoCluster`.
+    /// * `app_name` - The name of the app being managed, such as `hdfs` or `trino`.
+    /// * `role` - The role that this object belongs to, e.g. `datanode` or `worker`.
+    /// * `operator_name` - The DNS-style name of the operator managing the object (such as `hdfs.stackable.tech`).
+    /// * `controller_name` - The name of the controller inside of the operator managing the object (such as `hdfscluster`)
+    pub fn new_with_role<T: Resource<DynamicType = ()>>(
+        owner: &T,
+        app_name: &str,
+        role: &str,
+        operator_name: &str,
+        controller_name: &str,
+    ) -> OperatorResult<PodDisruptionBudgetBuilder<ObjectMeta, LabelSelector, ()>> {
+        let role_selector_labels = role_selector_labels(owner, app_name, role);
+        let metadata = ObjectMetaBuilder::new()
+            .namespace_opt(owner.namespace())
+            .name(format!("{}-{}", owner.name_any(), role))
+            .ownerreference_from_resource(owner, None, Some(true))?
+            .with_labels(role_selector_labels.clone())
+            .with_label(
+                APP_MANAGED_BY_LABEL.to_string(),
+                format_full_controller_name(operator_name, controller_name),
+            )
+            .build();
+
+        Ok(PodDisruptionBudgetBuilder {
+            metadata,
+            selector: LabelSelector {
+                match_expressions: None,
+                match_labels: Some(role_selector_labels),
+            },
+            ..PodDisruptionBudgetBuilder::default()
+        })
+    }
+
+    pub fn new_with_metadata(
```

Comment:
Let's also add a short doc comment here and below. It should shorty explain what is being set.